charts/rollout-operator/templates/webhook-role/mutatingwebhookconfiguration.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.webhooks.enabled -}}
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  labels:
+    grafana.com/inject-rollout-operator-ca: "true"
+    grafana.com/namespace: {{.Release.Namespace}}
+    {{- include "rollout-operator.labels" . | nindent 4 }}
+  name: prepare-downscale-{{.Release.Namespace}}
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: {{ include "rollout-operator.fullname" .}}
+      namespace: {{.Release.Namespace}}
+      path: /admission/prepare-downscale
+      port: 443
+  failurePolicy: Fail
```

Comment:
This may create a deadlock when applying it. If this webhook is created but rollout-operator isn't updated yet, then the rollout operator won't have the https endpoint and will always fail, rejecting any operation like trying to update the rollout operator.

It should be deployed with `Ignore` strategy first, and then with `Fail`. Note that `Ignore` only ignores failed calls, it doesn't ignore explicit rejections.

Reply:
is this only true for the `mutatingwebhookconfiguration.yaml` or does this also apply to the `validatingwebhookconfiguration.yaml`? 

is something like this:
```
 {{- if eq((lookup "v1" "MutatingWebhookConfiguration" "{{.Release.Namespace}}" ) "" ) }}
  failurePolicy: Ignore
  {{- else }}
  failurePolicy: Ignore
  {{- end }}
```
what you had in mind?

Reply:
This applies to both webhooks. 

I don't understand the condition of your diff (not saying it's wrong, I'm just not familiar with Helm), but I could imagine this as an extra setting, something to be disabled by default, and to be enabled after rolling out for more robustness (so it wouldn't allow everything if rollout-operator is just crashing).

Also, your diff says `Ignore` in both branches :D

Reply:
thanks for the update/clarification, i think i get what you're saying.